tests/components/elmax/test_config_flow.py

```diff
@@ -0,0 +1,169 @@
+
+    user_input = {
+        CONF_ELMAX_PANEL_NAME: MOCK_PANEL_NAME,
+        CONF_ELMAX_PANEL_PIN: MOCK_PANEL_PIN,
+    }
+    step_panels_result = await flow.async_step_panels(user_input=user_input)
+    assert step_panels_result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert step_panels_result["reason"] == "single_instance_allowed"
+
+
+async def test_invalid_credentials(hass):
+    """Test that invalid credentials throws an error."""
+    conf = {
+        CONF_ELMAX_USERNAME: "wrong_user_name@email.com",
+        CONF_ELMAX_PASSWORD: "incorrect_password",
+    }
+
+    flow = config_flow.ConfigFlow()
+    flow.hass = hass
+
+    with patch(
+        "elmax_api.http.Elmax.login",
+        side_effect=ElmaxBadLoginError(),
+    ):
+        result = await flow.async_step_user(user_input=conf)
+        assert result["errors"] == {"base": "bad_auth"}
+
+
+async def test_connection_error(hass):
+    """Test other than invalid credentials throws an error."""
+    conf = {CONF_ELMAX_USERNAME: MOCK_USERNAME, CONF_ELMAX_PASSWORD: "password"}
+
+    flow = config_flow.ConfigFlow()
+    flow.hass = hass
+
+    with patch(
+        "elmax_api.http.Elmax.login",
+        side_effect=ElmaxNetworkError(),
+    ):
+        result = await flow.async_step_user(user_input=conf)
+        assert result["errors"] == {"base": "network_error"}
+
+
+async def test_invalid_pin(hass):
+    """Test error is thrown when a wrong pin is used to pair a panel."""
+    flow = config_flow.ConfigFlow()
+    flow.hass = hass
+
+    # Perform first step (user) to setup ConfigFlow internal state objects.
+    user_input = {CONF_ELMAX_USERNAME: MOCK_USERNAME, CONF_ELMAX_PASSWORD: "password"}
+    await flow.async_step_user(user_input=user_input)
+
+    # Simulate bad pin response.
+    with patch(
+        "elmax_api.http.Elmax.get_panel_status",
+        side_effect=ElmaxBadPinError(),
+    ):
+        user_input = {
+            CONF_ELMAX_PANEL_NAME: MOCK_PANEL_NAME,
+            CONF_ELMAX_PANEL_PIN: "111111",
+        }
+        step_panels_result = await flow.async_step_panels(user_input=user_input)
+        assert step_panels_result["errors"] == {"base": "invalid_pin"}
+
+
+async def test_step_user(hass):
+    """Test that the user step works."""
+    flow = config_flow.ConfigFlow()
+    flow.hass = hass
+
+    # Try to add another instance of the integration for the very same panel.
+    user_input = {CONF_ELMAX_USERNAME: MOCK_USERNAME, CONF_ELMAX_PASSWORD: "password"}
+    step_user_result = await flow.async_step_user(user_input=user_input)
+    assert step_user_result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert step_user_result["step_id"] == "panels"
+
+    user_input = {
+        CONF_ELMAX_PANEL_NAME: MOCK_PANEL_NAME,
+        CONF_ELMAX_PANEL_PIN: MOCK_PANEL_PIN,
+    }
+    step_panels_result = await flow.async_step_panels(user_input=user_input)
+    assert step_panels_result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
```

Comment:
You should patch `async_setup_entry` to avoid trigger the platform setup.

Reply:
Not sure I understood. Can you be more specific about that?

Reply:
Platform setup is not triggered here, there is no need to patch this